lib/puppet/provider/grafana_organization/grafana.rb

```diff
@@ -1,97 +1,50 @@
 # frozen_string_literal: true
 
 require 'json'
+require 'erb'
 
 require File.expand_path(File.join(File.dirname(__FILE__), '..', 'grafana'))
 
 Puppet::Type.type(:grafana_organization).provide(:grafana, parent: Puppet::Provider::Grafana) do
   desc 'Support for Grafana organizations'
 
-  defaultfor kernel: 'Linux'
-
-  def organizations
-    response = send_request('GET', format('%s/orgs', resource[:grafana_api_path]))
-    raise format('Failed to retrieve organizations (HTTP response: %s/%s)', response.code, response.body) if response.code != '200'
-
-    begin
-      organizations = JSON.parse(response.body)
-
-      organizations.map { |x| x['id'] }.map do |id|
-        response = send_request 'GET', format('%s/orgs/%s', resource[:grafana_api_path], id)
-        raise format('Failed to retrieve organization %d (HTTP response: %s/%s)', id, response.code, response.body) if response.code != '200'
-
-        organization = JSON.parse(response.body)
-
-        {
-          id: organization['id'],
-          name: organization['name'],
-          address: organization['address']
-        }
-      end
-    rescue JSON::ParserError
-      raise format('Failed to parse response: %s', response.body)
+  # https://grafana.com/docs/grafana/latest/http_api/org/#get-organization-by-name
+  def get_org_by_name(org_name)
+    response = send_request('GET', format('%s/orgs/name/%s', resource[:grafana_api_path], ERB::Util.url_encode(org_name)))
+
+    case response.code
+    when '404'
+      org = nil
+    when '200'
+      org = JSON.parse(response.body)
+    else
+      raise format('Failed to retrieve organization %s (HTTP response: %s/%s)', org_name, response.code, response.body)
     end
+
+    org
   end
 
   def organization
-    @organization ||= organizations.find { |x| x[:name] == resource[:name] }
+    @organization ||= get_org_by_name(resource[:name])
     @organization
   end
 
   attr_writer :organization
 
-  def id
-    organization[:id]
-  end
-
-  def id=(value)
-    resource[:id] = value
```

Comment:
If this line of code was ever hit, `resource[:id]` would already equal `value`, and you can't update the database id of a grafana resource anyway.